odoo/tools/mail.py

```diff
@@ -167,7 +175,7 @@ def parse_style(self, el):
                 del el.attrib['style']
 
 
-def html_sanitize(src, silent=True, sanitize_tags=True, sanitize_attributes=False, sanitize_style=False, sanitize_form=True, strip_style=False, strip_classes=False):
+def html_sanitize(src, silent=True, sanitize_tags=True, sanitize_attributes=False, sanitize_style=False, sanitize_form=True, strip_style=False, strip_classes=False, strip_qweb=True):
```

Comment:
I suppose you want to differentiate html fields (rendered) frop raw qweb html fields (holding qweb instructions). Is it really necessary to differentiate them ? What are the use case to choose between keeping qweb tags and removing them ? Is it harmful to keep them in all html fields by default ?

My rationale behind this question is that adding options makes code easier to understand and maintain. Here we add yet another ``if/else`` that changes the way we parse and accept / reject html content, leading to potential confusions.

cc @mart-e for security / html

Reply:
True, I am fine removing the option and never strip qweb tags.

Reply:
Could you clarify why you thought it was useful? Where would that `strip_qweb` behavior be desired, and why?

I'm asking because this has security implications. HTML/JS and qweb/jinja2 are 2 different types of data, and we should never mix the evaluation contexts for each type.

Reply:
Before this commit, qweb tags and attributes were stripped as they were not whitelisted and it broke some desired behaviors with the html fields.
I thought to allow to turn on the stripping of qweb in case we would need to activate the same behavior as before for some sanitization.

I guess the question is `where do we need to not strip it ?`.

To answer this question I made a branch where qweb tags are always stripped:
- https://runbot.odoo.com/runbot/bundle/master-qweb-template-stripqweb-false-rtest-nby-86169
The first test that break is `test_mail_composer_mass_wtpl`.
The reason is that `composer_form.body != self.template.body_html`.
Because the `mail.template` `body_html` field is `sanitize=False` whereas `mail.composer.mixin`  `body` is `sanitize=True` (by default).

If I am not mistaken, the field `body` of `mail.composer.mixin` sould accept qweb now that it does not use jinja anymore. So either we specify `strip_qweb=False` or `sanitize=False` to make it work.

This is why `strip_qweb` is true by default as the argument of the function and false by default for html fields.
It would be even safer to turn `strip_qweb` to true by default for html fields (as it is the current behavior) and specifying `strip_qweb=False` for fields that should not strip it.

Reply:
> I guess the question is `where do we need to not strip it ?`.

Maybe you wanted to ask "why do we sanitize things?" first, to get a better understanding of what you are changing ;)

Sanitizing is used for fields that contain raw HTML that will be output unmodified in a context where it could become executable (e.g. in a qweb template with `t-raw`/`t-out`), or directly put into the DOM by some JS code. This is the default use case for many `fields.Html` cases, where we need to make the user input "trusted" before we inject it.

On the other hand, if the field contains QWeb code that is going to be evaluated, it is completely pointless to sanitize it. The contents is not final yet, and the QWeb expressions can add/remove anything in there when they will be evaluated, making it completely unsafe again.

For example, if you allow `t-raw`, then I can use `<span t-raw="record.name">` and put some `<script>` tag in the name of the record, and the result will be unsafe. Sanitizing that is useless.

For me you can drop `allowed_tags_qweb` and all that clique, it seems to be nonsense.

Reply:
You're right, thank you for the explanation.